### ISSUE TITLE
node-fastcgi compatibility with node ≥ 0.11.8

### ISSUE THREAD

**Issue by ebraminio** [joyent/node#8988]
I've filed this bug on [node-fastcgi](https://github.com/fbbdev/node-fastcgi/issues/2) bug tracker also. I think it could be a more general compatibility issue for next release of node so I am reporting it here also:

First have a look at node-fastcgi and the bug I've filed [there](https://github.com/fbbdev/node-fastcgi/issues/2). Here is my configurations:

.lighttpd.conf:
```conf
server.document-root = "."
server.modules += ( "mod_fastcgi" )
server.port = 2020
fastcgi.server += (
  "/myapp" => ((
    "socket" => "/tmp/myapp-fcgi.sock",
    "bin-path" => "/home/user/myapp/server.js",
    "check-local" => "disable",
    "max-procs" => 1,
  ))
)
```

[server.js](https://github.com/fbbdev/node-fastcgi#example):
```JavaScript
#!/home/user/.nvm/v0.11.8/bin/node

var fcgi = require('node-fastcgi');

fcgi.createServer(function(req, res) {
  if (req.method === 'GET') {
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end("It's working");
  } else {
    res.writeHead(501);
    res.end();
  }
}).listen();
```

Do `chmod +x server.js` and `npm install node-fastcgi` then `lighttpd -f .lighttpd.conf -D` and open http://127.0.0.1:2020/myapp on your browser. It works if you install 0.11.7 and change server.js to use it but for 0.11.8 and newer node versions, it just makes browser waiting forever and does nothing.

**Comment by cjihrig** [comment]
Thanks for reporting this. Let's see what comes from the issue you opened on node-fastcgi. If that module needs to be updated, then it should happen there. If you have an example that doesn't contain any external dependencies and demonstrates a problem with Node core, then we can handle it here.

**Comment by ebraminio** [comment]
Interestingly, by more compiling node/bisecting I've found that the bug is coming from libuv upgrade
* 613d76e works
* a35a212 doesn't compile
* 74a664b myapp is broken
So I think it is not just JSish toy code issue and I think this would be better to get more attention, (cc: @tjfontaine ) My naive guess is the issue is on handling unix sockets on newer version of libuv.

**Comment by ebraminio** [comment]
The issue is fixed on libuv but it will be shipped with [1.2.1](https://github.com/libuv/libuv/issues/124#issuecomment-69971509) and I heard node 0.12 will be shipped with libuv v1.0. Is it possible to fix the issue somehow so node-fcgi (and may a lot more clients that are using node as a child process) wouldn't break on transitioning to Node 0.12?

**Comment by saghul** [comment]
> I heard node 0.12 will be shipped with libuv v1.0.

How so? Would it help if I made a PR upgrading libuv to 1.2.1?

**Comment by ebraminio** [comment]
On other projects sometimes I see a patch is backported to a beta branch. I thought that can work here by reverting the patch caused the regression or the one fixes the issue.

**Comment by cjihrig** [comment]
@tjfontaine has it been decided what version of libuv will ship with 0.12?

**Comment by saghul** [comment]
https://github.com/joyent/node/pull/9050

**Comment by misterdjules** [comment]
Reopening and adding to the 0.12.1 milestone, as the second release candidate has already been released, and upgrading libuv to a version that fixes this issue has been determined to be too big a change at this point in the release process.

**Comment by misterdjules** [comment]
Fixed with #9179, which should be released with node v0.12.2 (planned tomorrow, Tuesday, but could slip).
